engine/src/main/java/org/hibernate/validator/internal/metadata/aggregated/NonContainerCascadingMetaData.java

```diff
@@ -95,6 +101,34 @@ public boolean isContainer() {
 		return false;
 	}
 
+	public ContainerCascadingMetaData getLegacyContainerCascadingMetaData(Class<?> valueClass) {
+		if ( List.class.isAssignableFrom( valueClass ) ) {
+			return new ContainerCascadingMetaData( List.class, List.class.getTypeParameters()[0], List.class, List.class.getTypeParameters()[0],
```

Comment:
Could we use singleton instances of these meta-data types perhaps? At least for the case where there are no group conversions?